tests/foreman/api/test_computeresource_gce.py

```diff
@@ -0,0 +1,357 @@
+    """Creates finish image on GCE Compute Resource"""
+    finish_image = entities.Image(
+        architecture=module_architecture, compute_resource=module_gce_compute,
+        name=gen_string('alpha'), operatingsystem=module_os,
+        username=finishuser, uuid=gce_latest_rhel_uuid
+    ).create()
+    return finish_image
+
+
+class TestGCEComputeResourceTestCase:
+    """Tests for ``api/v2/compute_resources``."""
+
+    @tier1
+    def test_positive_crud_gce_cr(self, module_org, module_location):
+        """Create, Read, Update and Delete GCE compute resources
+
+        :id: b324f8cd-d509-4d7a-b738-08aefafdc2b5
+
+        :expectedresults: GCE Compute resource should be created, read, updated and deleted
+
+        :CaseImportance: Critical
+
+        :CaseLevel: Component
+        """
+        cr_name = gen_string('alpha')
+        # Testing Create
+        compresource = entities.GCEComputeResource(
+            name=cr_name,
+            provider='GCE',
+            email=GCE_SETTINGS['client_email'],
+            key_path=GCE_SETTINGS['cert_path'],
+            project=GCE_SETTINGS['project_id'],
+            zone=GCE_SETTINGS['zone'],
+            organization=[module_org],
+            location=[module_location]
+        ).create()
+        assert compresource.name == cr_name
+        # Testing Update
+        new_name = gen_string('alpha')
+        description = gen_string('utf8')
+        new_zone = random.choice(VALID_GCE_ZONES)
+        compresource.name = new_name
+        compresource.description = description
+        compresource.zone = new_zone
+        compresource.update(['name', 'description', 'zone'])
+        # Testing Read
+        updated = entities.GCEComputeResource(id=compresource.id).read()
+        assert updated.name == new_name
+        assert updated.description == description
+        assert updated.zone == new_zone
+        # Testing Delete
+        updated.delete()
+        assert not entities.GCEComputeResource().search(
+            query={'search': 'name={}'.format(new_name)})
+
+    @pytest.mark.skip_if_open("BZ:1794744")
+    @tier3
+    def test_positive_check_available_images(self, module_gce_compute, googleclient):
+        """Verify all the images from GCE are available to select from
+
+        :id: 5cdfab18-a591-4442-8c19-a01e9b10ac36
+
+        :expectedresults: All the images from Google CR should be available to select in GCE CR
+
+        :CaseLevel: Integration
+        """
+        satgce_images = module_gce_compute.available_images()
+        gcloudclinet_images = googleclient.list_templates(True)
+        assert len(satgce_images) == len(gcloudclinet_images)
+
+    @tier3
+    def test_positive_check_available_networks(self, module_gce_compute, googleclient):
+        """Verify all the networks from GCE are available to select
+
+        :id: 259c9f5f-19dd-4e06-af5e-874612914f44
+
+        :expectedresults: All the networks from Google CR should be available to select in GCE CR
```

Comment:
Technically, this only tests if number of networks is the same, not if they are actually the same networks.

Does `module_gce_compute.available_networks()` provide list of networks as well? Perhaps you could put that into `set()` and compare against `set()` of `googleclient.list_network()`?